ui/hooks/useCarouselManagement/fetchCarouselSlidesFromContentful.ts

```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+import { CarouselSlide } from '../../../shared/constants/app-state';
+
+const SPACE_ID = process.env.CONTENTFUL_ACCESS_SPACE_ID;
+const ACCESS_TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
+const ENVIRONMENT = 'master'; // or 'dev'
+const CONTENT_TYPE = 'promotionalBanner';
+const DEFAULT_DOMAIN = 'cdn.contentful.com'; // or 'preview.contentful.com';
+const CONTENTFUL_API = `https://${DEFAULT_DOMAIN}/spaces/${SPACE_ID}/environments/${ENVIRONMENT}/entries`;
+
+export async function fetchCarouselSlidesFromContentful() {
+  const res = await axios.get(`${CONTENTFUL_API}`, {
+    params: {
+      access_token: ACCESS_TOKEN,
+      content_type: CONTENT_TYPE,
+    },
+  });
+
+  const assets = res.data.includes?.Asset || [];
+  const resolveImage = (imageRef: any) => {
```

Comment:
Hmm we have some type/linting errors here.
I'll see if I can help generate some Contentful types to help here.